File: front-end/financiamento-veiculos-study-web/src/app/shared/models/simulacao-financiamento-request.ts
import { ClienteRequest } from './cliente-request';
import { SimulacaoRequest } from './simulacao-request';

export class SimulacaoFinanciamentoRequest {
  clienteRequest: ClienteRequest;
  simulacaoRequest: SimulacaoRequest;

  constructor(cliente: ClienteRequest, simulacao: SimulacaoRequest) {
    this.clienteRequest = cliente;
    this.simulacaoRequest = simulacao;
  }
}

File: microservices/FinanciamentoVeiculosStudy/src/main/java/com/financiamentos/veiculosstudy/entity/endpoint/SimulacaoFinanciamentoRequest.java
package com.financiamentos.veiculosstudy.entity.endpoint;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.experimental.FieldNameConstants;

@Getter
@Builder
@AllArgsConstructor
@NoArgsConstructor
@FieldNameConstants
//@JsonNaming(PropertyNamingStrategy.SnakeCaseStrategy.class)
public class SimulacaoFinanciamentoRequest {
    private ClienteRequest clienteRequest;
    private SimulacaoRequest simulacaoRequest;
}

